libs/blocks/preflight/panels/performance.js

```diff
@@ -0,0 +1,272 @@
+  key: text.personalization.key,
+});
+
+export const checkVideosWithoutPosterAttribute = async () => conditionalItemUpdate({
+  failsWhen: !config.lcp.element.poster,
+  emptyWhen: !config.lcp.url.match('media_.*.mp4'),
+  key: text.videoPoster.key,
+});
+
+export const checkIcons = () => conditionalItemUpdate({
+  failsWhen: config.lcp.element.closest('.section').querySelector('.icon-milo'),
+  key: text.icons.key,
+});
+
+export function PerformanceItem({ icon, title, description }) {
+  return html` <div class="seo-item">
+    <div class="result-icon ${icon}"></div>
+    <div class="seo-item-text">
+      <p class="seo-item-title">${title}</p>
+      <p class="seo-item-description">${description}</p>
+    </div>
+  </div>`;
+}
+
+export const checkForSingleBlock = () => conditionalItemUpdate({
+  failsWhen: document.querySelector('main > div.section').childElementCount > 1,
+  key: text.singleBlock.key,
+});
+
+export const createPerformanceItem = ({
+  icon,
+  title,
+  description,
+} = {}) => html`<${PerformanceItem}
+  icon=${icon}
+  title=${title}
+  description=${description}
+/>`;
+
+let clonedLcpSection;
+function highlightElement(event) {
+  const lcpSection = config.lcp.element.closest('.section');
+  const tooltip = document.querySelector('.lcp-tooltip-modal');
+  const { offsetHeight, offsetWidth } = lcpSection;
+  const scaleFactor = Math.min(500 / offsetWidth, 500 / offsetHeight);
+  if (!clonedLcpSection) {
+    clonedLcpSection = lcpSection.cloneNode(true);
+    clonedLcpSection.classList.add('lcp-clone');
+    clonedLcpSection.style.width = `${lcpSection.offsetWidth}px`;
+    clonedLcpSection.style.height = `${lcpSection.offsetHeight}px`;
+    clonedLcpSection.style.transform = `scale(${scaleFactor})`;
+    clonedLcpSection.style.transformOrigin = 'top left';
+  }
+  if (!tooltip.children.length) tooltip.appendChild(clonedLcpSection);
+  const { top, left } = event.currentTarget.getBoundingClientRect();
+  tooltip.style.width = `${offsetWidth * scaleFactor}px`;
+  tooltip.style.height = `${offsetHeight * scaleFactor}px`;
+  tooltip.style.top = `${top + window.scrollY - offsetHeight * scaleFactor - 10}px`;
+  tooltip.style.left = `${left + window.scrollX}px`;
+  document.querySelector('.lcp-tooltip-modal').classList.add('show');
+}
+
+const removeHighlight = () => { document.querySelector('.lcp-tooltip-modal').classList.remove('show'); };
+
+function observePerfMetrics() {
+  // Observe LCP
+  const lcpObserver = new PerformanceObserver((entryList) => {
+    const entries = entryList.getEntries();
+    const lastEntry = entries[entries.length - 1];
+    if (lastEntry && !lastEntry.url.includes('preflight')) {
+      config.lcp = lastEntry;
+    }
+    const hasValidLcp = checkLCP();
+    if (!hasValidLcp) return;
+    checkFragments();
+    checkForPersonalization();
+    checkVideosWithoutPosterAttribute();
+    checkIcons();
+    checkForSingleBlock();
+    Promise.all([checkImageSize(), checkPlaceholders()]);
```

Comment:
This is a bit of a funky case.
- The hover for the element will be obviously
- None of the other checks are relevant if we don't have a good LCP candidate to begin with
![Screenshot 2024-10-17 at 01 24 12](https://github.com/user-attachments/assets/fedcf19b-fe30-4989-9a40-97c71ef78128)